app/main.py

```diff
@@ -1,3 +1,62 @@
+from typing import Any
+
+
 class Distance:
-    # Write your code here
-    pass
+
+    def __init__(self, km: (int, float)) -> None:
+        self.km = km
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __add__(self, other: Any) -> "Distance":
+        if isinstance(other, (int, float)):
+            return Distance(self.km + other)
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+
+    def __iadd__(self, other: Any) -> "Distance":
+        if isinstance(other, (int, float)):
+            self.km += other
+        if isinstance(other, Distance):
+            self.km += other.km
+        return self
+
+    def __mul__(self, other: (int, float)) -> "Distance":
+        return Distance(self.km * other)
+
+    def __truediv__(self, other: (int, float)) -> "Distance":
```

Comment:
Consider adding a check to handle division by zero in the `__truediv__` method to prevent runtime errors.